toolchain/check/testdata/interop/cpp/class/access.carbon

```diff
@@ -529,6 +529,204 @@ class D {
   }
 }
 
+// --- overload_set_base_class.h
+
+#include "overload_set.h"
+
+class PublicDerived : public C {};
+class ProtectedDerived : protected C {};
+class PrivateDerived : private C {};
+
+// --- import_overload_set_public_base_class_call_public.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "overload_set_base_class.h";
+
+fn F() {
+  //@dump-sem-ir-begin
+  Cpp.PublicDerived.foo();
+  //@dump-sem-ir-end
+}
+
+// --- fail_import_overload_set_public_base_class_call_non_public.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "overload_set_base_class.h";
+
+fn F() {
+  // CHECK:STDERR: fail_import_overload_set_public_base_class_call_non_public.carbon:[[@LINE+5]]:3: error: cannot access protected member `foo` of type `Cpp.PublicDerived` [ClassInvalidMemberAccess]
+  // CHECK:STDERR:   Cpp.PublicDerived.foo(1);
+  // CHECK:STDERR:   ^~~~~~~~~~~~~~~~~~~~~~~~
+  // CHECK:STDERR: fail_import_overload_set_public_base_class_call_non_public.carbon: note: declared here [ClassMemberDeclaration]
+  // CHECK:STDERR:
+  Cpp.PublicDerived.foo(1);
+  // CHECK:STDERR: fail_import_overload_set_public_base_class_call_non_public.carbon:[[@LINE+5]]:3: error: cannot access private member `foo` of type `Cpp.PublicDerived` [ClassInvalidMemberAccess]
+  // CHECK:STDERR:   Cpp.PublicDerived.foo(1, 2);
+  // CHECK:STDERR:   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
+  // CHECK:STDERR: fail_import_overload_set_public_base_class_call_non_public.carbon: note: declared here [ClassMemberDeclaration]
+  // CHECK:STDERR:
+  Cpp.PublicDerived.foo(1, 2);
```

Comment:
Where you're doing multiple calls to `foo`, do you think it'd be helpful to comment about what's distinct about each call? The name "foo" and the parameters "1" don't carry meaning, and the diagnostic doesn't point out the different overrides, so I worry that outside the context of a PR it'll be hard to understand.

A different approach might be to create types that make it clearer the difference between each overload (e.g., `struct PrivateCall {};`) so that you would see it in the signature (e.g. `Cpp.PublicDerived.foo(PrivateCall.PrivateCall());`).